### DISCUSSION TITLE
Implement custom migrations with data modeling

### DISCUSSION THREAD

**Discussion by bernatvadell** [directus/directus#6141] [Feature Requests]
Hi!

I'm just starting with Directus after working with other options.

In most projects, we usually offer multiple work environments before moving the application to production:

- Development
- Staging / QA
- Pre-production
- Production

I was looking for an extension that would allow me to execute the changes made in develop as migrations and propagate them through the different environments.

For example, Hasura has implemented that every time you make a change in the schema / data model, it generates a migration file (up / down) that I can easily integrate into GIT to automate the changes through CI / CD..

https://hasura.io/docs/latest/graphql/core/migrations/index.html

**Comment by rijkvanzanten** [comment]
This is all discussed in more detail over at https://github.com/directus/directus/discussions/3891 🙂
